### DISCUSSION TITLE
Users might want to use prompts rather than just queries

### DISCUSSION THREAD

**Discussion by yigit353** [assafelovic/gpt-researcher#861] [Ideas]
In systems like GTPR, the general theme is to help the user build their own prompts in the background by only asking for their query and adding settings like the length of the target document or the tone of the document to ensure consistency. Basically, this is why there is a tool in the first place.

However, sometimes, power users want to have more fine-tuned control over how the document is formatted. 

There could be many ways to achieve this. Perplexity's approach here is to allow users to add their own custom prompt, which is like adding an extra prompt to the general prompt. There is also the more advanced collections feature, which adds prompts that are local to these collections.

Another approach could be, as an initial step, to allow users to add a custom prompt before sending the query, which is not enabled by default. With a click, it can be opened. 

One of the main issue I detected here was sometimes I want GPTR to create the report just based on the documents I uploaded. I tried to tell the system by prompting, but it used retrieval instead. For this problem, it might be better to make it a feature rather than free-form prompting. Still, there could be other edge cases, for example, specific tones that couldn't be included in the list or specific style demands, which could be added via prompting.

The idea of using prompts instead of just queries could use a lot of brainstorming and can create many feature ideas continuously.

**Comment by sfdxb7** [comment]
I like this

**Comment by ElishaKay** [comment]
@assafelovic,

How about we add a 3rd tab on the Settings modal, called "Prompt Settings"?

There could be a long-text form there for filling out whatever the user wants to append their query with

**Reply by yigit353** [reply]
@ElishaKay The idea sounds like a good start. Which prompt is changed based on the user prompt? It may depend on the selected report type.
Also, users might want to jailbreak or do hacky things when provided with custom prompting, which might raise another concern.
